lib/errorResponseInterceptor.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * © Copyright IBM Corporation 2024. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function errorResponseInterceptor(axiosError) {
+  if (
+    axiosError?.response && // must have a response
+    axiosError?.response.status >= 400 &&
+    axiosError?.config.method !== 'head' &&
+    // must have data:
+    axiosError.response.data &&
+    // must be a JSON:
+    axiosError?.config.responseType === 'json' &&
```

Comment:
This again seems redundant. What does this give us that not covered by further checks on content type and instance of Object? Or other way around - if this passes, can we skip next two checks?